code/iaas/logic/src/main/java/io/cattle/platform/process/instance/InstanceRemove.java

```diff
@@ -48,7 +49,7 @@ protected void storage(Instance instance, Map<String, Object> data) {
             }
         }
 
-        List<Mount> mounts = getObjectManager().children(instance, Mount.class);
+        List<Mount> mounts = getObjectManager().find(Mount.class, MOUNT.REMOVED, null, MOUNT.INSTANCE_ID, instance.getId());
         for (Mount mount : mounts) {
```

Comment:
What if mount is in inactive/removing state? Do we want to exclude them as well?